src/data/ApiSettingsCodec.ts

```diff
@@ -1,5 +1,8 @@
 import { Codec, string, number, optional } from "purify-ts";
 
+const optionalPeriod = optional(Codec.interface({ day: number, month: number }));
+const optionalPeriodLastYear = optional(Codec.interface({ units: string, value: number }));
```

Comment:
Would it make sense to narrow "units" also at this point? then we can completely remove the function that validates/parses it.